cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -2635,4 +2763,120 @@ B";
         ";
         parse(YaccKind::Original(YaccOriginalActionKind::NoAction), src).unwrap();
     }
+
+    #[test]
+    fn test_self_describing_yacckind() {
+        let srcs = [
+            r#"%grmtools {
+  yacckind Original(NoAction)
+}
+%%
+S: "()";"#,
+            r#"
+%grmtools {
+  yacckind Original(UserAction)
+}
+%%
+S: "()";"#,
+            r#"%grmtools {
+  yacckind Grmtools
+}
+%start S
+%%
+S -> (): "()" { () };"#,
+        ];
+        for yacc_src in srcs {
+            parse(YaccKind::SelfDescribing(None), yacc_src).unwrap();
+        }
+
+      let fallback_srcs = [
+        r#"%start S
+           %%
+           S: "()";"#,
+        r#"%grmtools {
+           }
+           %start S
+           %%
+           S: "()";"#
+      ];
+      for yacc_src in fallback_srcs {
+            parse(YaccKind::SelfDescribing(Some(Box::new(YaccKind::Original(YaccOriginalActionKind::NoAction)))), yacc_src).unwrap();
+      }
+
+    }
+    #[test]
```

Comment:
Missing newline between functions.

Reply:
Should be fixed in [a307980](https://github.com/softdevteam/grmtools/pull/491/commits/a307980095753e5fa028e6de465eab90a7fd7e72)